### ISSUE TITLE
make test is broken

### ISSUE THREAD

**Issue by Chiiruno** [bakape/meguca#889]
```
go test --race -p 1 meguca/...
ok      meguca  1.094s
?       meguca/assets   [no test files]
ok      meguca/auth     (cached)
ok      meguca/cache    (cached)
ok      meguca/common   (cached)
ok      meguca/config   (cached)
psql: FATAL:  Peer authentication failed for user "meguca"
panic: exit status 2

goroutine 1 [running]:
meguca/db.init.1()
        /home/okina/go/src/github.com/bakape/meguca/go/src/meguca/db/init_test.go:9 +0xeb
FAIL    meguca/db       0.020s
?       meguca/geoip    [no test files]
psql: FATAL:  Peer authentication failed for user "meguca"
panic: exit status 2

goroutine 1 [running]:
meguca/imager.TestMain(0xc00018c200)
        /home/okina/go/src/github.com/bakape/meguca/go/src/meguca/imager/main_test.go:19 +0x21c
main.main()
        _testmain.go:70 +0x222
FAIL    meguca/imager   0.035s
ok      meguca/imager/assets    1.014s
?       meguca/lang     [no test files]
?       meguca/log      [no test files]
psql: FATAL:  Peer authentication failed for user "meguca"
panic: exit status 2

goroutine 1 [running]:
meguca/parser.init.1()
        /home/okina/go/src/github.com/bakape/meguca/go/src/meguca/parser/main_test.go:14 +0x15d
FAIL    meguca/parser   0.017s
2018-09-15 11:55:02.804622917-05:00   INFO dropping previous test database
psql: FATAL:  Peer authentication failed for user "meguca"
panic: exit status 2

goroutine 1 [running]:
meguca/server.init.1()
        /home/okina/go/src/github.com/bakape/meguca/go/src/meguca/server/router_test.go:34 +0x398
FAIL    meguca/server   0.042s
ok      meguca/templates        (cached)
?       meguca/test     [no test files]
ok      meguca/util     (cached)
2018-09-15 11:55:03.647013108-05:00   INFO dropping previous test database
psql: FATAL:  Peer authentication failed for user "meguca"
panic: exit status 2

goroutine 1 [running]:
meguca/websockets.init.0()
        /home/okina/go/src/github.com/bakape/meguca/go/src/meguca/websockets/websockets_test.go:47 +0x1de
FAIL    meguca/websockets       0.018s
2018-09-15 11:55:04.278123614-05:00   INFO dropping previous test database
psql: FATAL:  Peer authentication failed for user "meguca"
panic: exit status 2

goroutine 1 [running]:
meguca/websockets/feeds.init.2()
        /home/okina/go/src/github.com/bakape/meguca/go/src/meguca/websockets/feeds/thread_watcher_test.go:22 +0x147
FAIL    meguca/websockets/feeds 0.016s
make: *** [Makefile:94: test] Error 1
```

**Comment by bakape** [comment]
How did you create the test database?

**Comment by Chiiruno** [comment]
`sudo createdb -T template0 -E UTF8 -O meguca meguca_test`
Need to do as postgres user?

**Comment by bakape** [comment]
The problem is your current user is not called meguca and PG kinda expects
that for peer authentication. Best course of action would be to add a
config for specifying DB login parameters for tests.

**Comment by Chiiruno** [comment]
Okay. We'll need to update the docs.

**Comment by Chiiruno** [comment]
No, I'm retarded.
Keeping this open until the commit is added.

**Comment by bakape** [comment]
Want to handle it?

**Comment by Chiiruno** [comment]
I'd rather not, at least today.
#888 is pretty much all I'm going to be able to do today.

**Comment by bakape** [comment]
- [ ] Depends on #940 
- [ ] Adding a configuration to the JSON for the test DB user
  - [ ] Default will be current (get from os package) for PAM authentication
  - [ ] Mention user needs DB creation rights and how to get them in the development docs

**Comment by Robbotto** [comment]
Hi,

Please unsubscribe me to this.

Thank you.

Jaime

**Comment by bakape** [comment]
@Robbotto 
![screenshot_20181217_201914](https://user-images.githubusercontent.com/7851952/50106686-2fcf0a80-0239-11e9-9367-d3585daf22ad.png)

**Comment by Chiiruno** [comment]
lol what

**Comment by bakape** [comment]
@Chiiruno Can't you just edit `pg_hba.conf` and set the appropriate role authentication method to `trust`. It seems I did just that and then forgot.

**Comment by Chiiruno** [comment]
Can't seem to get that to work, even with ident mapping.
Please note I am using postgresql 10.

**Comment by bakape** [comment]
Post the file and whichever role you are using for meguca.

**Comment by Chiiruno** [comment]
I'm on NixOS, so it's a bit different, but it's more or less the default.
Meguca tries to do `make test` with the default postgres parameters, which you set as meguca, meguca, the problem is that my username is 'okina'.
I guess I could just `sudo -u meguca` for `make test`, though.
Perhaps the documentation needs a fool-proof way of setting up postgres for testing?
Otherwise, maybe the defaults need to be configurable.

**Comment by Chiiruno** [comment]
Geh, it's not that easy, no `sudo -u meguca`, then.

**Comment by bakape** [comment]
Did you create a PostgreSQL role `meguca` and give it DB creation rights?

**Comment by bakape** [comment]
Anyway, post your `pg_hba.conf`.

**Comment by Chiiruno** [comment]
I honestly can't find it, or it doesn't exist.
NixOS is weird.
That said, it should be this: https://github.com/NixOS/nixpkgs/blob/release-18.09/nixos/modules/services/databases/postgresql.nix#L178
I changed it to:
```
local all all trust
host all all ::1/128 trust
```

And was able to get more permissions, but I guess it's a role issue now with DB creation rights.

**Comment by bakape** [comment]
>NixOS is weird

Closing in favour of #980
